### PR TITLE
Basic post analytics

### PR DESCRIPTION
**Pull request by jpaddison3**
Adds a page for authors to view how popular their posts are, as measured by views. See: https://forum.effectivealtruism.org/posts/vLoKHGgyDom4Acine/pr-faq-sharing-readership-data-with-forum-authors.

NB: Relies on: https://github.com/centre-for-effective-altruism/ForumAnalytics/pull/11

image

Currently very minimal data, in order to get an MVP out. Next up would be graphs, and probably getting reading time into a better state.

This involved updating pg-promise, and converting the creation of analytics queries to be async.

### CODE REVIEW

File: packages/lesswrong/components/posts/PostsAnalyticsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        <em>Calculating metrics. (This can take some time for popular posts.)</em>
+      </Typography>
+      <Loading />
+    </>
+  }
+  if (error) {
+    throw error
+  }
+
+  return (<Table>
```
Comment (jcmustin) [inline packages/lesswrong/components/posts/PostsAnalyticsPage.tsx:49]: image

This looks a bit cramped on mobile. Layout.tsx switches from 50px top padding to 0px at the 'sm' breakpoint, which is most of the problem. I think adding top and side padding would both help.

Related, I feel like that's a weird default behavior for Layout.tsx to have. I feel like the default padding should be some sensible nonzero value, and any pages that need 0 top padding should use negative padding to compensate. Not a blocking issue for this PR though.

File: packages/lesswrong/components/posts/PostsPage/PostsPagePostHeader.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { extractVersionsFromSemver } from '../../../lib/editor/utils'
 import { getUrlClass } from '../../../lib/routeUtil';
 import classNames from 'classnames';
 import { isServer } from '../../../lib/executionEnvironment';
+import EditIcon from '@material-ui/icons/Edit'
```
Comment (jcmustin) [inline packages/lesswrong/components/posts/PostsPage/PostsPagePostHeader.tsx:9]: Looks like this is unused here.

File: packages/lesswrong/components/posts/PostsAnalyticsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    fragmentName: 'PostsPage',
+  })
+  const currentUser = useCurrentUser()
+  const serverRequestStatus = useServerRequestStatus()
+  const {
+    SingleColumnSection, WrappedLoginForm, PostsAnalyticsInner, HeadTags, Typography
+  } = Components
+
+  if (!query.postId) {
+    return <SingleColumnSection>
```
Comment (jcmustin) [inline packages/lesswrong/components/posts/PostsAnalyticsPage.tsx:85]: Minor point: should this update the request status like the cases below it?

```suggestion
    if (serverRequestStatus) serverRequestStatus.status = 400
    return <SingleColumnSection>
```

**Review by jcmustin — APPROVED** [review]
I'm excited about this! I left a few small comments, but otherwise things look good to me. Nicely done!

File: packages/lesswrong/components/posts/PostsAnalyticsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  // want 30 seconds before TTFB
+  return <>
+    <HeadTags title={title} />
+    <SingleColumnSection>
+      <Typography variant='body1' className={classes.gutterBottom}>{title}</Typography>
+      <NoSsr>
+        <PostsAnalyticsInner post={post} />
+      </NoSsr>
+        <Typography variant="body1" className={classes.viewingNotice} component='div'>
+        <p>This features is new. <Link to='/contact-us'>Let us know what you think.</Link></p>
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsAnalyticsPage.tsx:120]: Small typo? "This feature is new"

**Pull request merged**